packages/ember-metal/lib/mixin.ts

```diff
@@ -348,14 +350,16 @@ function replaceObserversAndListeners(
   obj: object,
   key: string,
   prev: Function | null,
-  next: Function | null
+  next: Function | null,
+  seen: Set<Function>
 ): void {
   if (typeof prev === 'function') {
     updateObserversAndListeners(obj, key, getObservers(prev), removeObserver);
     updateObserversAndListeners(obj, key, getListeners(prev), removeListener);
   }
 
-  if (typeof next === 'function') {
+  if (typeof next === 'function' && !seen.has(next)) {
```

Comment:
with `seen` avoiding same `function` being passed through `updateObserversAndListeners` since same function might be used multiple times `observer` `event listeners`